### PR TITLE
chore: use `node:` protocol imports

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes

[Use `node:` protocol][1] to import builtin modules

## Context

`node:` protocol imports are now the preferred way to import builtin Node.js modules. Advantages are:
- It's immediately clear where an import is coming from
- Zero chance of something in `node_modules` overriding an import
- Going forward new Node.js modules will only be available via `node:` style imports, starting with [`node:test` in Node.js v18][2]

An `eslint-plugin-import` rule is currently under consideration to enforce this. See https://github.com/import-js/eslint-plugin-import/issues/2717

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

[1]: https://nodejs.org/api/esm.html#node-imports
[2]: https://nodejs.org/docs/latest-v18.x/api/test.html

### CODE REVIEW

**Review by rarkins — COMMENTED** [review]
Q1: Looks like they're still sorted first?
Q2: No problem for us generating a CJS output?

**Comment by JamieMagee** [comment]
> Q1: Looks like they're still sorted first? 

Yeah, no issues with our eslint-enforced sorting

> Q2: No problem for us generating a CJS output?

Yep. Node has supported them in CJS since v14.18.0 and v16.0.0, and in ESM since v12.20.0, v14.13.1, and v16.0.0

File: test/fixtures.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import type fs from 'fs';
-import type { PathLike, Stats } from 'fs';
+import type fs from 'node:fs';
+import type { PathLike, Stats } from 'node:fs';
```
Comment (rarkins) [inline test/fixtures.ts:2]: Can these be combined?

File: lib/modules/platform/bitbucket-server/utils.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // SEE for the reference https://github.com/renovatebot/renovate/blob/c3e9e572b225085448d94aa121c7ec81c14d3955/lib/platform/bitbucket/utils.js
-import url from 'url';
+import url, { URL } from 'node:url';
```
[the remark this answers is not in the document]
Comment (JamieMagee) [reply lib/modules/platform/bitbucket-server/utils.ts:2]: Yes, `url`refers to the legacy URL API[^1]. `URL` is the new [WHATWG URL API][2]. We are currently using both in this file. We were using the implicit `URL` global object, but I added `URL` here so we are explicitly using the new WHATWG URL API.

[2]: https://nodejs.org/api/url.html#the-whatwg-url-api
[^1]: https://nodejs.org/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost

[the remark this answers is not in the document]
Comment (JamieMagee) [reply lib/modules/platform/bitbucket-server/utils.ts:2]: Removing the legacy `url` usage is outside the scope of this PR. But I can revert the explicit `URL` reference.

File: lib/modules/datasource/docker/index.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import URL from 'url';
+import URL from 'node:url';
```
Comment (viceice) [inline lib/modules/datasource/docker/index.ts:1]: we should get rid of this deprecated module

File: lib/modules/platform/bitbucket-server/utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (viceice) [reply lib/modules/platform/bitbucket-server/utils.ts:2]: importing `URL` from `url` is discouraged, the whole module is deprecated. using the global is the preferred usage.

**Review by viceice — APPROVED** [review]
otherwise LGTM

File: lib/modules/datasource/docker/index.ts (hunk 3 above)
Comment (JamieMagee) [reply lib/modules/datasource/docker/index.ts:1]: Tracked in #7797

File: lib/modules/platform/bitbucket-server/utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (viceice) [reply lib/modules/platform/bitbucket-server/utils.ts:2]: can't find it any longer. i think it was inside the nodejs docs. but for better compabillity we should prefer the global object. as we probably like to run renovate inside the browser in future ☺️

**Pull request merged**